Added a getStatusCode method.

o0Ignition0o opened the pull request:
this should fix #27 

Please let me know if there's anything I can do to improve the pull request :)

luzlab commented:
Can we please merge this or PR29 to get the typescript enums?

prettymuchbryce commented:
My only issue with this change is that currently this project is not compiled. This means using features like `Array.find` and `Object.entries` without a polyfill could break browser compatibility.

https://caniuse.com/#feat=array-find

o0Ignition0o commented:
Oh I understand, do you think using a for loop instead is the way to go ? if so i can edit it today :)

o0Ignition0o commented:
Using a for loop + rebased as required :)

prettymuchbryce commented:
This looks better. We still need to add this to the TS definition file, and probably some mention in the README as well.

o0Ignition0o commented:
Oh yeah sure, I will add it today. Thanks a lot for providing a bit of
mentoring despite a tight schedule !

On Thu, Mar 7, 2019, 23:53 Bryce Neal <notifications@github.com> wrote:

> This looks better. We still need to add this to the TS definition file,
> and probably some mention in the README as well.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/prettymuchbryce/http-status-codes/pull/28#issuecomment-470732698>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJBvTq_L8M-E-iSTcqSWfhu__1h6iV1iks5vUZhlgaJpZM4ZAwcA>
> .
>

o0Ignition0o commented on index.js:
```diff
@@ -74,3 +74,13 @@ exports.getStatusText = function(statusCode) {
+exports.getStatusCode = function(statusText) {
+  for (key in statusCodes) {
+    if (statusCodes[key] === statusText) {
+      return parseInt(key, 10);
```
Do we want to be more defensive here and check if parseInt is a number?

prettymuchbryce replied:
I think it's fine. Lets assume `key` is always some Number (status code) coerced into a string.

prettymuchbryce commented on README.md:
```diff
@@ -26,6 +26,9 @@ response
 		error: HttpStatus.getStatusText(HttpStatus.INTERNAL_SERVER_ERROR)
 	});
+
+response.status(HttpStatus.getStatusCode('No Content'))
```
Can you format these with the line breaks like the others?

prettymuchbryce commented on README.md:
```diff
@@ -107,14 +110,22 @@ response
 	})
+
+response
+	.status(getStatusCode('No Content')
```
How about we just use internal server error like the example above?

prettymuchbryce commented on index.d.ts:
```diff
@@ -384,5 +384,13 @@ export declare enum HTTP_STATUS {
  * Convert the numeric status code to its appropriate title.
  * @param statusCode One of the available status codes in this package
  * @returns {String} The associated title of the passed status code
+ * @throws {Error} The status code does not exist
```
Nice catch. 👍

prettymuchbryce commented on index.d.ts:
```diff
@@ -384,5 +384,13 @@ export declare enum HTTP_STATUS {
+ * Convert the status code title to its appropriate numeric value
+ * @param statusText One of the available status texts in this package
+ * @returns {Number} The associated status code of the passed status text
+ * @throws {Error} The status text does not exist
```
I think you forgot to close this comment.

prettymuchbryce commented on index.d.ts:
```diff
@@ -384,5 +384,13 @@ export declare enum HTTP_STATUS {
 export declare function getStatusText(statusCode: number): string;
+
+/**
+ * Convert the status code title to its appropriate numeric value
```
I have some problem with the word "title" in these comments. We should refer to them consistently as "status code" and "status text". 

In fact the IETF spec calls it a "Reason Phrase" so ideally we could change all of the comments from "title" -> "reason phrase".

```
The reason-phrase element exists for the sole purpose of providing a
   textual description associated with the numeric status code, mostly
   out of deference to earlier Internet application protocols that were
   more frequently used with interactive text clients.  A client SHOULD
   ignore the reason-phrase content.
```

o0Ignition0o commented:
Thanks a lot for such a thorough code review. I think I have addressed most of the concerns :)

By the way I was skimming over the HTTP status codes and it seems like the 500 Reason phrase is ["Internal Server Error"](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.5.1) whereas we here refer to it as ["Server Error"](https://github.com/prettymuchbryce/http-status-codes/blob/master/README.md#codes).

As stated in the IETF spec, a client should ignore the reason-phrasse content so it's probably not a big deal, plus we probably wouldn't want to cause a breaking change ^^

justinmchase commented:
Can we merge this now please?

prettymuchbryce commented:
Sorry for the delay.
